crypto/bio/file.c

```diff
@@ -193,6 +198,13 @@ static long file_ctrl(BIO *b, int cmd, long num, void *ptr) {
       b->shutdown = (int)num & BIO_CLOSE;
       b->ptr = ptr;
       b->init = 1;
+#if defined(OPENSSL_WINDOWS)
+      // Windows differentiates between "text" and "binary" file modes, so set
+      // the file to text mode if caller specifies BIO_FP_TEXT flag.
+      //
+      // https://learn.microsoft.com/en-us/cpp/c-runtime-library/reference/setmode?view=msvc-170#remarks
+      _setmode(_fileno(b->ptr), num & BIO_FP_TEXT ? _O_TEXT : _O_BINARY);
```

Comment:
OpenSSL also [makes a call to `setvbuf` on Windows](https://github.com/openssl/openssl/blob/84364b9dc693a30fa55c22e684b45978a5bcc77b/crypto/bio/bss_file.c#L238C1-L246C55) when `BIO_FP_TEXT` is set.  Should we do the same?
```
            /*
             * Reports show that ftell() isn't trustable in text mode.
             * This has been confirmed as a bug in the Universal C RTL, see
             * https://developercommunity.visualstudio.com/content/problem/425878/fseek-ftell-fail-in-text-mode-for-unix-style-text.html
             * The suggested work-around from Microsoft engineering is to
             * turn off buffering until the bug is resolved.
             */
            if ((num & BIO_FP_TEXT) != 0)
                setvbuf((FILE *)ptr, NULL, _IONBF, 0);
```

Reply:
Another good question. I'll write some test cases to determine whether it's needed.

Reply:
Based on the reproduction test I've written (specifically, that it passes without the call to `setvbuf` on our windows CI), I don't think we need this call. I'd like to avoid porting over OpenSSL's legacy cruft for dealing with fixed bugs.